src/python/pants/option/options_fingerprinter.py

```diff
@@ -84,6 +79,11 @@ def fingerprint(self, option_type, option_val):
     if option_val is None:
       return None
 
+    # The python documentation (https://docs.python.org/2/library/json.html#json.dumps) states that
+    # dict keys are coerced to strings in json.dumps, but this appears to be incorrect.
+    if isinstance(option_val, dict):
```

Comment:
Seems wise to not use `collections.{Iterable,Mapping}` above or else also use those here and below.

Reply:
Sorry about that. I will use the `collections` types. Not trying to forget your comments.

Reply:
Using `collections.Iterable` here causes the pre-commit hook to fail (even after I remembered that a Mapping is Iterable too), but it should almost definitely be used and if tests pass with the most recent commit I will figure out how to do that.

Reply:
This is the code I used to get that error (in the `fingerprint()` method body):
```python
    # For simplicity, we always fingerprint a list.  For non-list-valued options,
    # this will be a singleton list.
    if not isinstance(option_val, collections.Iterable):
      # The python documentation (https://docs.python.org/2/library/json.html#json.dumps) states
      # that dict keys are coerced to strings in json.dumps, but this appears to be incorrect, so we
      # need to encode them ourselves.
      if isinstance(option_val, collections.Mapping):
        option_val = {stable_json_sha1(k):v for k, v in option_val.items()}

      option_val = [option_val]
```